gitlab/cmd/receive_adapter/main.go

```diff
@@ -0,0 +1,81 @@
+/*
+Copyright 2020 The TriggerMesh Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package main
+
+import (
```

Comment:
can we here use adapter.Main(), like discussed here:
https://knative.dev/development/eventing/samples/writing-receive-adapter-source/05-receive-adapter/

and in other sources, like: https://github.com/knative/eventing-contrib/blob/master/prometheus/cmd/receive_adapter/main.go

Reply:
from what I see, adapter.Main() logic is based on a poll loop implemented in Start(stopCh) method. I don't think that poll loop is suitable for a webhooks which are waiting for incoming events, that's why we tried to stick to the githubsource approach:
https://github.com/knative/eventing-contrib/blob/c6d009760096ad14bad4d383af4e75d97cb1df6f/github/cmd/receive_adapter/main.go#L110-L129

Please let me know if there are a way to implement webhooks using adapter.Main()

Reply:
There should not be any polling involved. You Server should simply be blocking within `Start()`.

We implemented something similar in Kyma, feel free to have a look:
* [shared main](https://github.com/kyma-project/kyma/blob/master/components/event-sources/cmd/http-adapter/main.go) invokes `NewAdapter()` to initialize an adapter.
* [NewAdapter()](https://github.com/kyma-project/kyma/blob/master/components/event-sources/adapter/http/adapter.go#L67) returns a type that implements `adapter.Adapter`.
* [Start()](https://github.com/kyma-project/kyma/blob/master/components/event-sources/adapter/http/adapter.go#L85) runs a server (CloudEvents receiver), which blocks until termination.

Reply:
thanks for the explanation!
@antoineco could you take a look at the updated [receive_adapter.go](https://github.com/knative/eventing-contrib/blob/95f3e24e26fca80daffd873d08912b30b1555643/gitlab/pkg/adapter/receive_adapter.go)? If it's ok I update tests

Reply:
Looks good to me! 👍 
So you decided to not use `webhooks.Run()` in the end?

Reply:
yes, not only because webhooks.v5 package don't have `Run()` method, but also because it provides more control over the execution :slightly_smiling_face:

Reply:
Fully agree